include/GloryDuckTables2019Lkl.h

```diff
@@ -0,0 +1,74 @@
+//////////////////////////////////////////////////////////////////////
+// Binned flux likelihood
+//////////////////////////////////////////////////////////////////////
+
+#ifndef GLORYDUCKLKL
+#define GLORYDUCKLKL
+
+#include "TObjArray.h"
+#include "TCanvas.h"
+
+#include "Lkl.h"
+
+class GloryDuckTables2019Lkl : public Lkl
+{
+ public:
+  // constructors
+  GloryDuckTables2019Lkl(TString inputString="");
+  
+  // destructor
+  virtual ~GloryDuckTables2019Lkl();
+ 
+  // make the TObjArray own its contents
+  void SetOwner(Bool_t set=kTRUE)            {fSampleArray->SetOwner(set);}
+
+  // add samples
+  inline void AddSample(Lkl* sample)        {fSampleArray->Add(sample); SetChecked(kFALSE);}
+
+  // getters
+  inline const TObjArray* GetSampleArray()     const {return fSampleArray;}
+  inline       Int_t      GetNSamples()        const {return fSampleArray->GetEntries();}
+  inline       Lkl*       GetReferenceSample() const {return (Lkl*)fSampleArray->At(0);}
+  inline       Lkl*       GetSample(Int_t i)   const {return (Lkl*)fSampleArray->At(i);}
+  inline       Double_t   GetLogJ()            const {return fLogJ;}
+  inline       UInt_t     GetNMasses()         const {return fNMasses;}
+  inline       UInt_t     GetActiveMass()      const {return fActiveMass;}
+
+  // unable SetUnitsOfG, dPhi/dE_signal must have proper units
+  virtual void SetUnitsOfG(Double_t unit);
+
+  // set mass index to the current active one
+  void SetActiveMass(Double_t mass);
+
+  // print data in the overview
+  virtual void PrintData(Int_t level=0);
+
+  // plots
+  TCanvas* PlotInputData();
+
+  // data input
+  Int_t ReadGloryDuckInputData(TString inputfilename);
+
+ protected:
+          Int_t InterpretInputString(TString inputString);
+  virtual void  SetFunctionAndPars(Double_t ginit=0);
+  virtual Int_t MakeChecks();
+  virtual void  SetMinuitLink();
+  inline  void  ClearSampleList()  {fSampleArray->Clear();}
+
+  // data input
+  Int_t CreateAndAddNewParabola(Double_t mass,Double_t svmin,Double_t svmax,Int_t npoints,Double_t* sv,Double_t* logL);
+
+ private:
+  UInt_t      fNMasses;       //   Number of masses
+  UInt_t      fActiveMass;    //   Position in the array fMass of the current active mass
+  TObjArray*  fSampleArray;   //-> Array of Lkl objects (one for each mass) 
+  Double_t*   fsvMin;         //-> [fNMasses] array with <sv> lower values
```

Comment:
I think we need a list of sv values, which is valid for all masses, right? that reflects the format we decided, doesn't it?
so not this line nor the following but rather 
Double_t* fsvVals

Reply:
While revising the .cc file, I realize that the list of <sv> values is stored in every ParabolaLkl object. Then I think you do not need to keep it also here, you can always access the list of <sv> values accessing any of the parabolas.
Maybe you want to add a function to the GloryDuckTables2019 class that acts as a shortcut to access the list of <sv> values from one of the Parabolas objects (e.g. the first one).

This assumes that all parabolas have the same <sv> list, which is enforced at the definition of the format only, but not in the implementation of the class. Maybe at the level of implementation we don't want to assume it, to be more flexible in view of possible changes of format. 

Al alternative to force that all <sv> values for all Parabolas are the same is to store the list of <sv> only once at the GloryDuckTables2019 level and in the Parabola terms, in the x axis instead of <sv> you just store consecutive integers from 0 to nSvVals-1

Reply:
I think that you do not need the fsvVals either, because those values are stored in every Parabola, is there any need to have them again saved as data members of GloryDuck class?